src/Common/MoistThermodynamics/profiles.jl

```diff
@@ -0,0 +1,284 @@
+#### Reference state profiles
+
+export VirtualTemperature,
+       FixedLapseRate,
+       IsothermalProfile,
+       DryAdiabaticProfile,
+       LinearTemperatureProfile
+
+"""
+    AbstractReferenceState{FT<:AbstractFloat}
+
+Reference states which can then be called with the functor:
+
+```
+(ref_state::AbstractReferenceState)(param_set::AbstractParameterSet, z::FT) {where FT<:AbstractFloat}
+```
+where
+ - `param_set` parameter set, used to dispatch planet parameter function calls
+ - `z` vertical (altitude) coordinate
+or
+ - `e_pot` potential energy (e.g., gravitational potential)
+"""
+abstract type AbstractReferenceState{FT<:AbstractFloat} end
+
+"""
+    FixedLapseRate{FT}
```

Comment:
Seems unnecessary to me to keep this here if we do not keep it as a reference profile in the hydrostatic state.

Reply:
We can/will remove it. I only included this for comparison with what we used to have (which we should be able to see in the docs)